### PR TITLE
.pickle format for Table, reorganization of file writers and readers

### PR DESCRIPTION
**Pull request by janezd**
I don't know how to modify the mock in one of the tests, so I disabled the test. @astaric , is there an elegant way to solve this?

### CODE REVIEW

**Comment by astaric** [comment]
Something like this?

    @patch("os.path.exists", Mock(return_value=True))
    def test_read_data_calls_reader(self):
        table_mock = Mock(data.Table)
        reader_instance = Mock(read_file=Mock(return_value=table_mock))

        with patch.dict(io.FILE_READERS, {'.xlsx': lambda: reader_instance}):
            table = data.Table.from_file("test.xlsx")

        reader_instance.read_file.assert_called_with("test.xlsx", data.Table)
        self.assertEqual(table, table_mock)

File: Orange/data/table.py [hunk 1]
```diff
@@ -469,18 +477,16 @@ def from_file(cls, filename):
...
-
+        reader = io.FILE_READERS.get(ext)
+        if not reader:
+            desc = io.FORMAT_NAMES.get(ext)
+            if desc:
+                raise IOError("Reading {}s is not supported".
+                              format(desc.lower()))
+            else:
+                raise IOError("Unknown file name extension.")
+        print(absolute_filename)
```
Comment (astaric) [inline Orange/data/table.py]: Please remove.

File: Orange/data/io.py [hunk 2]
```diff
@@ -617,3 +631,24 @@ def read_file(self, file, cls=None):
...
+    @staticmethod
+    def read_file(file, _=None):
+        return pickle.load(open(file, "rb"))
+
+    @staticmethod
+    def write_file(filename, table):
+        pickle.dump(table, open(filename, "wb"))
+
+
+FILE_FORMATS = (TabDelimFormat, TxtFormat, BasketFormat, ExcelFormat,
```
Comment (astaric) [inline Orange/data/io.py]: I would prefer a class decorator for this, so the registration of the format is closer  to the class definition.

Comment (janezd) [inline Orange/data/io.py]: @astaric Brilliant. Done.

**Pull request merged**